webview-ui/src/components/menu/Navbar.tsx

```diff
@@ -22,7 +23,13 @@ export const Navbar = () => {
 				name: "Chat",
 				tooltip: "New Task",
 				icon: PlusIcon,
-				navigate: navigateToChat,
+				navigate: () => {
+					// Close the current task, then navigate to the chat view
+					TaskServiceClient.clearTask({}).catch((error) => {
+						console.error("Failed to clear task:", error)
+					})
+					navigateToChat()
```

Comment:
Do we need to worry about race condition where we navigate to chat before the task was cleared?
```suggestion
					TaskServiceClient.clearTask({}).catch((error) => {
						console.error("Failed to clear task:", error)
					}).finally(() => navigateToChat())
```